packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -146,6 +148,17 @@ const DomainPicker: FunctionComponent< Props > = ( {
 		isExpanded ? quantityExpanded : quantity
 	);
 
+	const isValidCurrentDomain =
+		currentDomain?.domain_name !== existingSubdomain && currentDomain?.domain_name !== '';
+	const persistentDomainIsInSuggestions = domainSuggestions?.some(
+		( suggestion ) => suggestion.domain_name === persistentSelectedDomain?.domain_name
+	);
+
+	if ( isValidCurrentDomain && persistentSelectedDomain && ! persistentDomainIsInSuggestions ) {
+		// Append our currently selected domain to the suggestions so it's persistently visible to the user.
+		domainSuggestions?.push( persistentSelectedDomain );
```

Comment:
It's possible that I'm missing something but why not append directly `currentDomain` here so `persistentSelectedDomain` and the [useEffect below](https://github.com/Automattic/wp-calypso/pull/47633/files#diff-c3d188f313a48ead116f33025dc8b34ef32ac6db67a47629b1fa058cc3386383R170-R179) won't be needed?

Reply:
@razvanpapadopol You possibly could have been viewing some outdated changes. I have just pushed some updates to this logic.

RE `persistentSelectedDomain` We need to store the domain selection in component state so that when the user:

1. Selects a domain from the expanded selection in the Domain Details view
2. Goes back to summary view, and then picks their sub-domain for whatever reason
3. Now we need to keep their domain (from step 1) here so that it doesn't disappear following a rerender as this looks buggy and confusing if it does.

Reply:
@razvanpapadopol RE my point above about storing the domain in state, it is to meet the A/C of https://github.com/Automattic/wp-calypso/issues/47540

*"...the additional domain item should not disappear from the Domain Picker until the Summary view is re-visited."*